### DISCUSSION TITLE
Testing SSE connections

### DISCUSSION THREAD

**Discussion by infogulch** [Orange-OpenSource/hurl#2636] [Q&A]
I'd like to test a scenario where one connection makes an [SSE request](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events) and waits for messages that are triggered by a POST request made on a different connection.

**Comment by infogulch** [comment]
When I google this issue in regards to Curl itself, most of the answers mention using the `-N` option (no buffering), which is relevant in a cli context where the response would just print out to the terminal.

As far as Hurl is concerned, special support for SSE would be cool I guess. To avoid building in specific features for SSE, I think my case could be covered with two features:

1. Akin to a "no buffering" mode, a hurl request mode where it continuously tests the response predicates against the buffered response so far, and as soon as a success/failure determination occurs then terminate the connection (i.e. don't wait for the server to 'finish' the response).
    * If the test is `body contains "abc"` which could be anywhere in the response, then if it's not present yet it's success status is indeterminate and you just have to wait longer
    * If the test is `body startsWith "xyz"` and the first bytes of the request are `XYZ` then you can fail the test right away
    * I guess you'd want a test like `duration < 1000` to depend on the success status of the other asserts. If there are any indeterminate asserts then then it would wait until 1s then fail the test, but if all of the other asserts are success (with boolean implication logic if there are none) then it succeeds immediately.
2. Some kind of parallelism that is more fine-grained than whole files. Like, "start this request, but don't wait for it to complete before starting the next request". We already have [control flow options](https://hurl.dev/docs/entry.html#control-flow) like `skip` and `delay`, so maybe a new option like `async: true/false`?

Reading back, I'm doubtful if this would be *easier* than adding specific SSE support, but maybe there's other protocols that this would enable testing. 😅

**Comment by sjsadowski** [comment]
Popping in over two years later, adding SSE support would greatly reduce the complexity of my integration testing. I don't have much more to add other than currently I work around this by using curl directly with -N and --max-time, but adding support for long-running connections to hurl would be awesome.

**Comment by jcamiel** [comment]
Hi @sjsadowski 
Good SSE is definitely on our TODO, it will happens (no roadmap still but we know it's important)
